src/vm/edge_shape.rs

```diff
@@ -55,6 +55,12 @@ pub trait Edge: Copy + Send + Debug + PartialEq + Eq + Hash {
     fn prefetch_store(&self) {
         // no-op by default
     }
+
+    /// Create a edge from an address.
+    ///
+    /// Arguments:
+    /// *   `address`: The address in memory where an `ObjectReference` is stored.
+    fn from_address(address: Address) -> Self;
```

Comment:
I don't think we can turn any `Address` into an `Edge`. If we don't assume the layout of an object reference array, and try scan it in mmtk-core, we don't need this method.

Reply:
Removed. I introduced a new trait `MemorySlice`, similar to `Edge`, to solve this issue.